jupyterhub/handlers/base.py

```diff
@@ -429,10 +434,22 @@ async def get_current_user(self):
                 # don't let errors here raise more than once
                 self._jupyterhub_user = None
                 self.log.exception("Error getting current user")
-        if self._jupyterhub_user is not None or self.get_current_user_oauth_token():
-            self.scopes = roles.get_subscopes(*self._jupyterhub_user.roles)
         return self._jupyterhub_user
 
+    def _resolve_scopes(self):
+        self.raw_scopes = set()
+        app_log.debug("Loading and parsing scopes")
+        api_token = self.get_token()
+        if api_token:
+            self.raw_scopes = scopes.get_scopes_for(api_token)
+        elif self.current_user:
+            self.raw_scopes = scopes.get_scopes_for(self.current_user)
+        else:  # deprecated oauth tokens
+            user_from_oauth = self.get_current_user_oauth_token()
+            self.raw_scopes = scopes.get_scopes_for(user_from_oauth)
```

Comment:
This is where we should hardcode *different* scopes for oauth to preserve current behavior (read:user for just the owner), e.g. `self.raw_scopes = {'read:user': {'user': user_from_oauth.name}`. This here would expand oauth token permissions to match default api token permissions.

After #3380, this will go away, of course.